graph/src/schema/input/sqlexpr.rs

```diff
@@ -0,0 +1,323 @@
+            Array(_) => self.nope("Array"),
+            Interval(_) => self.nope("Interval"),
+            MatchAgainst { .. } => self.nope("MatchAgainst"),
+            Wildcard => self.nope("Wildcard"),
+            QualifiedWildcard(_) => self.nope("QualifiedWildcard"),
+        }
+    }
+
+    fn visit_func(&mut self, func: &mut p::Function) -> Result<(), ()> {
+        let p::Function {
+            name,
+            args: pargs,
+            filter,
+            null_treatment,
+            over,
+            distinct: _,
+            special: _,
+            order_by,
+        } = func;
+
+        if filter.is_some() || null_treatment.is_some() || over.is_some() || !order_by.is_empty() {
+            return self.illegal_function(format!("call to {name} uses an illegal feature"));
+        }
+
+        let idents = &mut name.0;
+        if idents.len() != 1 {
+            return self.illegal_function(format!(
+                "function name {name} uses a qualified name with '.'"
+            ));
+        }
+        self.visitor.visit_func_name(&mut idents[0])?;
+        for arg in pargs {
+            use p::FunctionArg::*;
+            match arg {
+                Named { .. } => {
+                    return self.illegal_function(format!("call to {name} uses a named argument"));
+                }
+                Unnamed(arg) => match arg {
+                    p::FunctionArgExpr::Expr(expr) => {
+                        self.visit_expr(expr)?;
+                    }
+                    p::FunctionArgExpr::QualifiedWildcard(_) | p::FunctionArgExpr::Wildcard => {
+                        return self
+                            .illegal_function(format!("call to {name} uses a wildcard argument"));
+                    }
+                },
+            };
+        }
+        Ok(())
+    }
+
+    fn check_binary_op(&mut self, op: &p::BinaryOperator) -> Result<(), ()> {
+        use p::BinaryOperator::*;
+        match op {
+            Plus | Minus | Multiply | Divide | Modulo => Ok(()),
+            StringConcat
+            | Gt
+            | Lt
+            | GtEq
+            | LtEq
+            | Spaceship
+            | Eq
+            | NotEq
+            | And
+            | Or
+            | Xor
+            | BitwiseOr
+            | BitwiseAnd
+            | BitwiseXor
+            | DuckIntegerDivide
+            | MyIntegerDivide
+            | Custom(_)
+            | PGBitwiseXor
+            | PGBitwiseShiftLeft
+            | PGBitwiseShiftRight
+            | PGExp
+            | PGOverlap
+            | PGRegexMatch
+            | PGRegexIMatch
+            | PGRegexNotMatch
+            | PGRegexNotIMatch
+            | PGLikeMatch
+            | PGILikeMatch
+            | PGNotLikeMatch
+            | PGNotILikeMatch
+            | PGStartsWith
+            | PGCustomBinaryOperator(_) => {
+                self.not_supported(format!("binary operator {op} is not supported"))
+            }
```

Comment:
Is explicitly listing th operators here instead of using `_` to force us to handle the case when a new operator is added in the future ?

Reply:
Yes, exactly. If upstream adds another operator, we'll get a build failure here and need to think about what we want to do with the new operator.